pkg/minishift/registration/redhat.go

```diff
@@ -51,17 +52,36 @@ func (registrator *RedHatRegistrator) CompatibleWithDistribution(osReleaseInfo *
 }
 
 func (registrator *RedHatRegistrator) Register(param *RegistrationParameters) error {
-	if output, err := registrator.SSHCommand("sudo -E subscription-manager version"); err != nil {
+	output, err := registrator.SSHCommand("sudo -E subscription-manager version")
+	if err != nil {
 		return err
-	} else {
-		if strings.Contains(output, "not registered") {
+	}
+	if strings.Contains(output, "not registered") {
+		for i := 1; i < 4; i++ {
+			if param.Username == "" {
+				param.Username = param.GetUsernameInteractive("Username to register to RHSM")
+			}
+			if param.Password == "" {
+				param.Password = param.GetPasswordInteractive("Password to register to RHSM")
+			}
 			subscriptionCommand := fmt.Sprintf("sudo -E subscription-manager register --auto-attach "+
 				"--username %s "+
 				"--password '%s' ", param.Username, util.EscapeSingleQuote(param.Password))
-			if _, err := registrator.SSHCommand(subscriptionCommand); err != nil {
+			_, err = registrator.SSHCommand(subscriptionCommand)
+			if err == nil {
+				return nil
+			}
+			if strings.Contains(err.Error(), "Invalid username or password") {
```

Comment:
So the idea is that if registration fails with explicitly specified credentials (env or CLI), it switches into an interactive mode? I guess that works. @LalatenduMohanty wdyt?